### PR TITLE
add value verification to check if device priorities value is available

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - add value verification for the key ov::device::priorities
 - add corresponding test case

### Tickets:
 - CVS-84419

### CODE REVIEW

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -806,7 +799,45 @@ std::string MultiDeviceInferencePlugin::GetDeviceList(const std::map<std::string
 
     return allDevices;
 }
-
+std::vector<std::string > MultiDeviceInferencePlugin::ParsePrioritiesDevices(std::string priorities, const char separator) {
```
Comment (songbell) [inline src/plugins/auto/plugin.cpp:802]: can we avoid pass by value

Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp]: updated.

File: src/plugins/auto/plugin.cpp [hunk 2]
```diff
@@ -806,7 +799,45 @@ std::string MultiDeviceInferencePlugin::GetDeviceList(const std::map<std::string
...
+            return false;
+        }
+        realDevName = DeviceIDParser(realDevName).getDeviceName();
+        std::string::size_type realEndPos = 0;
+        if ((realEndPos = realDevName.find('(')) != std::string::npos) {
+            realDevName = realDevName.substr(0, realEndPos);
+        }
+        if (_availableDevices.end() ==
+            std::find_if(_availableDevices.begin(), _availableDevices.end(), [&realDevName] (const std::string& device) {
+                return device.find(realDevName) != std::string::npos;
```
Comment (songbell) [inline src/plugins/auto/plugin.cpp:820]: something minor: or maybe don;t handle "(", just return realDevName.find(device)  != npos, is this what you expected?

File: src/plugins/auto/plugin.cpp [hunk 3]
```diff
@@ -891,12 +922,17 @@ void MultiDeviceInferencePlugin::CheckConfig(const std::map<std::string, std::st
...
+                return device.find(kvp.first) != std::string::npos;
+            })) {
             // keep secondary prperties for HW or virtual device
             continue;
         } else if (supported_configKeys.end() ==
                    std::find(supported_configKeys.begin(), supported_configKeys.end(), kvp.first)) {
             IE_THROW() << "Unsupported config key: " << kvp.first;
+        } else if (kvp.first == ov::device::priorities) {
+            // check if the device in ov::device::priorities is available.
+            ParsePrioritiesDevices(kvp.second);
```
Comment (songbell) [inline src/plugins/auto/plugin.cpp:935]: can you help explain the usage of this?

File: src/plugins/auto/plugin.cpp [hunk 4]
```diff
@@ -806,7 +789,42 @@ std::string MultiDeviceInferencePlugin::GetDeviceList(const std::map<std::string
 
     return allDevices;
 }
-
+std::vector<std::string > MultiDeviceInferencePlugin::ParsePrioritiesDevices(const std::string& priorities, const char separator) {
```
Comment (peterchen-intel) [inline src/plugins/auto/plugin.cpp:792]: Why not use MetaParseDevice directly?

Comment (yangwang201911) [reply src/plugins/auto/plugin.cpp:792]: deadlock happened when using `parseMetaDevice()` to check if device name in the priorities is available. call trace as below.
```
Core ie;
ie->compile_model(network, "AUTO", configs);
ie->GetCPPpluginByName("AUTO");
hold the pluginMutex and create AUTO plugin.
AUTO plugin::SetConfig(configs);
AUTO Plugin::CheckConfig();
AUTO Plugin::parseMetaDevices(ov::device::priorities.value());
GetCore()->GetAvaiableDevices();
ie->GetListOfDevicesInRegistry();
hold the pluginMutex again and deadlock happened here.
```

**Review by wangleis — APPROVED** [review]

File: src/plugins/auto/utils/config.hpp [hunk 5]
```diff
@@ -202,6 +202,18 @@ struct PluginConfig {
...
+                                                     "GNA",
+                                                     "TEMPLATE",
+                                                     "MYRIAD",
+                                                     "HDDL",
+                                                     "VPUX",
+                                                     "MULTI",
+                                                     "HETERO",
+                                                     "CUDA",
+                                                     "HPU_GOYA",
+                                                     "mock"};
```
[the remark this answers is not in the document]
Comment (nkogteva) [reply src/plugins/auto/utils/config.hpp:218]: > why do we need to keep hardcoded list of deivces?

@ilya-lavrenov @yangwang201911 The same question. I'm currently testing PROXY scenario and faced with 
```
C++ exception with description "Unavailable device name: Intel_GPU" thrown in the test body.
```

Please remove this hardcode.

[the remark this answers is not in the document]
Comment (yangwang201911) [reply src/plugins/auto/utils/config.hpp:218]: WIP for this.

**Pull request merged**